### ISSUE TITLE
Building Related Issues and [Feature Requests].

### ISSUE THREAD

**Issue by AdamDN** [SmartlyDressedGames/Unturned-3.x-Community#2652]
Hello! 
This is a list of building related issues which have existed in the game for years. I'm sure some of them have been reported before.
Most of them are extremely minor, but I've created a list of the most notable ones.
Every single one of these issues has affected the building I've just built, demonstrating that these issues do affect players on a daily basis. Both of the features requests would have also helped with the build.

**Issues**

1) The biggest and most annoying issue: 
Structures are not aligned when they snap. They always snap a couple of pixels off, and those pixels add up making it impossible to build large structures. 
<img width="1262" alt="5" src="https://user-images.githubusercontent.com/71527679/121798685-a63cff80-cc1f-11eb-87b3-c4182f3c7475.png">

2) The logs in wooden ramp walls are not equally spaced.
<img width="656" alt="1" src="https://user-images.githubusercontent.com/71527679/121797530-a7b6f980-cc18-11eb-84ec-485c78433f8f.png">

3) Wooden Ramp Walls require pillars to be placed, despite looking like they should only require posts.
(not really an issue, just a inconvenience and oddity)
<img width="569" alt="2" src="https://user-images.githubusercontent.com/71527679/121797570-eea4ef00-cc18-11eb-926c-7dbcbb1f4855.png">

4) Pillars can be placed inside posts, but posts cannot be placed inside pillars. 
<img width="440" alt="4" src="https://user-images.githubusercontent.com/71527679/121797634-67a44680-cc19-11eb-8808-53ac6f65a751.png">
<img width="427" alt="3" src="https://user-images.githubusercontent.com/71527679/121797636-696e0a00-cc19-11eb-8e8a-203ab4923c99.png">

5) Storages can clip through walls when they're placed too close. This happens with all barricades and is mostly unavoidable, but with storages it seems a little more important as they can be raided from outside and items can be grabbed through walls. 
<img width="566" alt="7" src="https://user-images.githubusercontent.com/71527679/121798083-282b2980-cc1c-11eb-86d4-e77be2278d92.png">

6) Text can still float off of signs when formatted in certain ways.
<img width="568" alt="10" src="https://user-images.githubusercontent.com/71527679/121799681-5b25eb00-cc25-11eb-9607-c583c292ca42.png">

7) Doors can (sometimes) be blocked by diagonally adjacent walls. This is related to issue 1, if the wall is even slightly misaligned it will block the door. 
![6](https://user-images.githubusercontent.com/71527679/121798792-4abf4180-cc20-11eb-9135-a266c2830753.gif)

**Feature Requests**

1) Right Angle Roofs. I'm not sure why these aren't in the game already, given that there is a plate variant. It would add tons of building possibilities both decoratively and defensively. 
<img width="1015" alt="8" src="https://user-images.githubusercontent.com/71527679/121798901-d638d280-cc20-11eb-8c94-a8cf10a5127e.png">

2) Resize Tool from map editor added to Building mode. This would be an extremely powerful addition, allowing for large floors and walls to be made at a fraction of the element count to reduce lag on servers - as well as adding infinitely more decoration possibilities.
For example in the screenshot below, 380 roofs could be replaced with a single element -- and would look nicer too. 
<img width="560" alt="9" src="https://user-images.githubusercontent.com/71527679/121798911-e2249480-cc20-11eb-9f95-60e48f1e3d6d.png">

**Comment by SDGNelson** [comment]
Unfortunately most of these are byproducts of "structures" functioning similar to "barricades" in that they are placed without any knowledge of their surroundings or connections. This should be improved as part of the base building performance improvements outlined at the start of the year, and from:
https://trello.com/c/auvkzChQ/30-base-building
#1520

5, 6, and feature request 2 are not likely to happen anytime soon however.

**Comment by AdamDN** [comment]
5 and 6 are understandable. 5 would just bring other complaints if storages were harder to place, and 6 is just an incredibly minor issue. 

Out of curiosity, is the feature request 2 unlikely to be implemented due to resized objects potentially being unstable, the tool being too powerful, or is it simply a low priority addition?


> This should be improved as part of the base building performance improvements outlined at the start of the year

As you mentioned the building performance improvements, I realize I forgot to mention the "lag lines" as one of the major building related issues.
I know there was a large blog post about them at the end of last year with the goal was to remove them by the end of this year. Is there any updates on the progress now that we're half way through the year?

**Comment by SDGNelson** [comment]
For structures resizing would be incompatible with the plan for tracking connections, and would add the overhead of storing and replicating scale for a very niche use.

The netcode rewrites so far this year were a big part of the lag line fix plan outlined in the blog post. I have not worked specifically on buildings recently, but still think it will be done for this year.

**Comment by SDGNelson** [comment]
Closing because the issues list was not really intended for feature requests. Sorry to close this discussion - I do WANT to fulfill all (most?) requests, but it is a matter of prioritization. Constantly being reminded of all the pending requests when checking issues is stressful. I will keep the request in mind. On GitHub maybe the Ideas section of Discussions would be the best place.
